prover/src/logup_gkr/prover.rs

```diff
@@ -147,7 +150,7 @@ fn build_mls_from_main_trace_segment<E: FieldElement>(
         match oracle {
             LogUpGkrOracle::CurrentRow(index) => {
                 let col = main_trace.get_column(*index);
-                let values: Vec<E> = col.iter().map(|value| E::from(*value)).collect();
+                let values: Vec<E> = iter!(col).map(|value| E::from(*value)).collect();
                 let ml = MultiLinearPoly::from_evaluations(values);
                 mls.push(ml)
             },
```

Comment:
I'm curious if iterating over oracles would work a bit better. For example:
```rust
for oracle in iter!(oracles) {
    ...
}
```

Also, a general question (not for this PR): we seem to be proactively converting base elements to extension field elements. Maybe there is a way to avoid it and only do this when we actually need to?

Reply:
I have answered in the other PR related to building MLEs.
For the second point, we have an issue to address that in #303

Reply:
For the first point, it is addressed in the other PR for building MLEs.
For the second point, we already have an issue for it in #303